jdbc-driver/src/main/java/com/google/cloud/dataproc/jdbc/HiveUrlUtils.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.dataproc.jdbc;
+
+import static com.google.cloud.dataproc.jdbc.DataprocDriver.DATAPROC_JDBC_HIVE_URL_SCHEMA;
+
+import java.io.IOException;
+import java.net.URI;
+
+/** Helper methods to establish Connection for DataprocDriver. */
+public class HiveUrlUtils {
+    public static final String HIVE_DEFAULT_DATABASE = "";
+    public static final int HIVE_DEFAULT_PORT = 10001;
+    public static final String URL_JDBC_PREFIX = "jdbc:";
+
+    /**
+     * Parses client url and extracts Hive connection parameters.
+     *
+     * @param url verified url passed in by client, already of valid format possible formats:
+     *     jdbc:dataproc://hive/<dbName>;clusterName=<>;projectId=<>;region=<>;port=<>
+     *     jdbc:dataproc://hive/<dbName>;clusterName=<>;projectId=<>;region=<>;clusterPoolLabel=foo=bar:env=staging;
+     * @return HiveJdbcConnectionParam object with fields correctly set
+     * @throws IOException
+     * @throws InvalidURLException
+     */
+    public static HiveJdbcConnectionOptions parseHiveUrl(String url) throws InvalidURLException {
+        if (url.equals(DATAPROC_JDBC_HIVE_URL_SCHEMA)) {
+            throw new InvalidURLException(
+                    String.format("Error URL: %s \n" + "Please provide cluster information.", url));
+        }
+
+        HiveJdbcConnectionOptions.Builder paramBuilder = HiveJdbcConnectionOptions.builder();
+
+        String urlStripped = url.substring(URL_JDBC_PREFIX.length());
+
+        URI jdbcURI = URI.create(urlStripped);
+
+        // getPath will return: /<dbName>;<sessionConfs>
+        String dbAndSessionConfs = jdbcURI.getPath().substring(1);
+
+        // no dbName or no cluster info
+        if (dbAndSessionConfs.indexOf(";") == -1) {
```

Comment:
There are two places where you're checking whether a cluster is specified (here and after the `switch` block).

You might be able to simplify a bit by just setting options on `HiveJdbcConnectionOptions` in each part of the switch statement, and then pushing validation logic into `HiveJdbcConnectionOptions#build()`: https://github.com/google/auto/blob/master/value/userguide/builders-howto.md#validate

Reply:
I now reduced the checking for whether a cluster (or cluster pool) is specified to only happen once, which is after the switch block. I feel that it's not necessary to make `clusterName, projectId, region, filter` to all be fields for `HiveJdbcConnectionOptions`, because essentially all we need is the host field, especially that `projectId, region, filter` only serve to be parameters for the Client Libraries to query the cluster info.
I plan to call the api after the switch block in `HiveUrlUtils`, rather than in `HiveJdbcConnectionOptions`, and check if 1) a `clusterName` is specified or 2) a cluster name can be correctly found using the api. If both no, then throw an exception.
From Karthik: You could theoretically make an autovalue object for those parameters if that makes sense. Or turn `HiveUrlUtils` into a class
Plan to include this change in next PR though.